### ISSUE TITLE
`route.action()` should not be called if route does not changes

### ISSUE THREAD

**Issue by manolo** [vaadin/vaadin-router#355]
When user pushes a link navigating to the current path, router should not call `navigate()` again. It should do nothing.

This is critical in the case of Flow because it makes a round trip to server side, and makes the server create a new view instance.

**Comment by vlukashov** [comment]
Could you please check if this is substantially different from https://github.com/vaadin/vaadin-router/issues/311? If the two issues are close enough, let's close this as a duplicate and use the earlier issue for tracking work.

**Comment by vlukashov** [comment]
This is sufficiently different from for #250 (because that's is about parameterized routes) and #311 (because that is about the component defined with the `component` route property whereas this issue is about the route `action`).

It's OK for the Router to keep calling `route.action()` on consequent navigation even when the URL does not change (because `action` might depend on an external state like `isUserAuthenticated`).

Vaadin Router should be able to handle the case when it gets exactly the same DOM element from a route `action` on consequent calls:
 - it should not call any lifecycle callbacks on the DOM element
 - it should not detach / re-attach the element to the DOM
 - basically, router should end resolution at that point
